corehq/form_processor/backends/couch/processor.py

```diff
@@ -159,7 +159,8 @@ def hard_rebuild_case(domain, case_id, detail, save=True, lock=True):
 
         assert case.domain == domain
         try:
-            if lock_obj:
+            if not found and lock_obj:
+                # the lock was only aquired if we found a case
                 acquire_lock(lock_obj, degrade_gracefully=False)
```

Comment:
Ah, now I see what you're doing. What do you think of moving this line up to line 159?